src/runez/file.py

```diff
@@ -279,7 +279,7 @@ def parent_folder(path, base=None):
     return path and os.path.dirname(resolved_path(path, base=base))
 
 
-def readlines(path, first=None, errors="ignore", fatal=False, logger=False):
+def readlines(path, first=None, errors="ignore", fatal=False, logger=False, rstrip=True):
```

Comment:
We can do this in the signature instead of `rstrip` boolean: `transform=str.rstrip`
Accept basically `transform` as an optional `callable`, and then:
```
line = decode(line)
if transform is not None:
    line = transform(line)

yield line
```

This will allow for more flexibility with this function (one can provide any `transform` of type `Callable[[str], str]`), and one can turn it off with `transform=None`